Ta routering update examples e overdracht (#63)
nuts-foundation/nl-generic-functions-ig: input/pagecontent/routing.md

The initial reason for drawing up this generic technical agreement is the need for a mechanism to address the notification, as described in the TA Notified Pull, at the right place within the receiving organization. Addressing the notification at the level of a healthcare organization isn’t sufficient for end users. 

#### Goal, scope, and principles

##### Goal
The mechanism defined in this document is intended to provide a single and simple mechanism of routing labels (persons, sub-organizations, locations, products, and services) within an organization and to allow organizations to define such routing labels.

##### Scope
This TA applies to scenarios where information exchange involves:  
- Publishing (sub-)organizations, practitioners and locations as routing labels in an mCSD Directory 
- Selecting a (sub-)organization, and practitioner or location from a structured directory using activity-types/definitions in healthcare services 
- Creating (clinical) requests and (logistic) tasks for the (sub-)organization, and practitioner or location 

> - Sending notifications to the corresponding endpoint of the (sub-)organization, and practitioner or location 

Reviewer: Klopt dit nog wel?

| Sending Organization   | The organization or party sending a message or request to the receiving organization                |
| Sending System         | The system sending a message or request on behalf of the sending organization                       |
| ServiceRequest         | A request for a service, such as a diagnostic procedure, surgery or other treatment to be performed on a patient  |
| System                 | Software used by (a) healthcare organization(s), such as an electronic health record or API service |
| Task                   | Workflow related activity in FHIR                                                    |
{:.grid .table-hover}

### Routing interactions
This chapter describes all relevant interactions on data level. 

#### mCSD-Based Directory Lookup
Routing is based on the IHE mCSD model, in which each care provider publishes Organization, Location, HealthcareService, and Endpoint resources in a mCSD Directory.  In this mCSD Directory, a sender can search for the appropriate service (e.g. “initial consultation” for “radiology” near “Zwolle”). See this [example content of a Query Directory](./Bundle-query-directory.html).

> Users (or their systems) first determine ***what*** they'd want to order or refer to patient for. The HealthcareService.type in the mCSD Directory contains one or more orderable items or service types. These can be represented as a (simple) code or as a ActivityDefinition/PlanDefinition. When a selection has been made, the request resource is created (using either a simple `.code` or `.instantiatesCanonical` in the case of an ActivityDefinition/PlanDefinition). See [example](./ServiceRequest-53a41e63-e826-45fa-9076-9be4b18399c8.html) Servicerequest. 

Reviewer 2: ```suggestion
Users (or their systems) first determine ***what*** they'd want to order for or refer the patient to. The `HealthcareService.type` in the mCSD Directory contains one or more orderable items or service types. These can be represented as a (simple) code or as an ActivityDefinition/PlanDefinition. When a selection has been made, the request resource is created (using either a simple `.code` or `.instantiatesCanonical` in the case of an ActivityDefinition/PlanDefinition). See [example](./ServiceRequest-53a41e63-e826-45fa-9076-9be4b18399c8.html) Servicerequest. 
```

> The FHIR Task is used to determine ***who*** and ***where*** to request should be delivered. The location is specified in `Task.location` and healthcareservice should be linked in `Task.owner` like in this [example](./Task-a0fc5221-bcd9-46f1-922f-c2913dae5d63.html).

Reviewer 2: ```suggestion
The FHIR Task is used to determine to ***who*** and ***where*** the request should be delivered. The location is specified in `Task.location` and the healthcare service should be linked in `Task.owner` like in this [example](./Task-a0fc5221-bcd9-46f1-922f-c2913dae5d63.html).
```

Users (or their systems) first determine ***what*** they'd want to order or refer to patient for. The HealthcareService.type in the mCSD Directory contains one or more orderable items or service types. These can be represented as a (simple) code or as a ActivityDefinition/PlanDefinition. When a selection has been made, the request resource is created (using either a simple `.code` or `.instantiatesCanonical` in the case of an ActivityDefinition/PlanDefinition). See [example](./ServiceRequest-53a41e63-e826-45fa-9076-9be4b18399c8.html) Servicerequest. 

The FHIR Task is used to determine ***who*** and ***where*** to request should be delivered. The location is specified in `Task.location` and healthcareservice should be linked in `Task.owner` like in this [example](./Task-a0fc5221-bcd9-46f1-922f-c2913dae5d63.html).

##### Role of the ServiceRequest
The ServiceRequest represents the ‘clinical intent’ or ‘clinical authorization’ for a procedure. It shall have a ServiceRequest.code (representing e.g. ‘initial consultation’ or ‘nursing handoff’) and may be instantiated by an ActivityDefinition or PlanDefinition

##### Role of the ActivityDefinition

> Using ActivityDefinitions directories can define more precisely how a service needs to be requested or distinguish between different activities that can be provided by a single service. Using PlanDefinitions (consisting of multiple ActivityDefinitions) allows for defining more complex workflows. 

Reviewer 2: The sentence seemed off, could it be that we need to place the ActivityDefinitions?

```suggestion
Using ActivityDefinitions in mCSD directories can define more precisely how a service needs to be requested or distinguish between different activities that can be provided by a single service. Using PlanDefinitions (consisting of multiple ActivityDefinitions) allows for defining more complex workflows. 
```

When the service directory defines an ActivityDefinition for the requested service, the server may decline a ServiceRequest for that service that does not refer to that ActivityDefinition in ServiceRequest.instantiatesCanonical or does not adhere to the requirements specified in that definition. 

##### Role of the Task
The Task is used coordination, i.e. state management and administrative routing (I.e ‘who’ and ‘where’ should do the referred ServiceRequest). The Task.code is used to define what the Task.owner is expected to do with the referred ServiceRequest, e.g. ‘fulfill’, ‘change’ or ‘approve’. Task.owner should refer to the HealthcareService responsible for carrying out the Request

##### Role of the HealthcareService
The HealthcareService is used to publish supported activities (using codeable concepts in element `.type` and optionally references to ActivityDefinitions and PlanDefinitions in element `.type.extension:supportedActivityDefinitions` ). These items are primarily used in the Request. 

> The HealthcareService.location may point to (multiple) physical locations where the service can be performed. If the HealthcareService does not specify a location, the location can be inherited from the organization (or, recursively, parent organizations) that provides the service. 

Reviewer 2: ```suggestion
The `HealthcareService.location` may point to (multiple) physical locations where the service can be performed. If the HealthcareService does not specify a location, the location can be inherited from the organization (or, recursively, parent organizations) that provides the service. 
```
